python/lib/modeldata/dmod/modeldata/data/object_store_dataset.py

```diff
@@ -314,24 +381,32 @@ def add_data(self, dataset_name: str, **kwargs) -> bool:
         """
         if dataset_name not in self.datasets:
             return False
-        elif 'file' in kwargs and 'directory' in kwargs:
+        elif data is not None:
+            result = self._client.put_object(bucket_name=dataset_name, data=data, length=len(data),
+                                             object_name=self._gen_dataset_serial_obj_name(dest))
+            return result.bucket_name == dataset_name
+        elif source is None or len(source) == 0:
             from sys import _getframe
-            msg = "{}.{} does not support both 'file' and 'directory' kwargs in a single call"
+            msg = "{}.{} requires either binary data or a source for data to be provided."
             raise ValueError(msg.format(self.__class__.__name__, _getframe(0).f_code.co_name))
-        elif 'file' in kwargs:
-            bucket_root = kwargs['bucket_root'] if 'bucket_root' in kwargs else kwargs['file'].parent
-            result = self._push_file(bucket_name=dataset_name, file=kwargs['file'], bucket_root=bucket_root)
-            # TODO: test
-            return isinstance(result.object_name, str)
-        elif 'directory' in kwargs:
-            bucket_root = kwargs['bucket_root'] if 'bucket_root' in kwargs else kwargs['directory']
+
+        src_path = Path(source)
+        if not src_path.exists():
+            from sys import _getframe
+            msg = "{}.{} source path '{}' does not exist."
+            raise ValueError(msg.format(self.__class__.__name__, _getframe(0).f_code.co_name, source))
+        elif src_path.is_dir():
+            bucket_root = kwargs['bucket_root'] if 'bucket_root' in kwargs else src_path
```

Comment:
`bucket_root = kwargs.get('bucket_root', src_path)`

Reply:
Fine, fine, I'll fix it.